### ISSUE TITLE
DataStore Multi Auth

### ISSUE THREAD

**Issue by dnys1** [aws-amplify/amplify-flutter#815]
**Describe the bug**

Feature request: Multiple authorization modes in DataStore, as supported in [iOS](https://docs.amplify.aws/lib/datastore/setup-auth-rules/q/platform/ios/#configure-multiple-authorization-types) and [Android](https://docs.amplify.aws/lib/datastore/setup-auth-rules/q/platform/android/#configure-multiple-authorization-types).

**Comment by matthewapps** [comment]
Hi, I've been wrestling with an auth issue for Flutter and datastore. This is the thread #9290[https://github.com/aws-amplify/amplify-js/issues/9290](url). Please can you help?

**Comment by leadsizzle** [comment]
Hi @dnys1 , kindly would like to know if there could be any information on the approximate date range for multi-auth datastore support to be released in the next 12 months or so? I'm following this feature as well. Very much thanks in advance and I also appreciate everyone's performance here, you guys are great.

**Comment by dorontal** [comment]
It would also be nice to also have an easy way to reconfigure Amplify at run-time, so that we can more easily take advantage of multi-auth. This is already possible in Javascript: to make witch your app to IAM mode, we use
```javascript
                    Amplify.configure({
                        aws_appsync_authenticationType: 'AWS_IAM'
                    });
```
or for Cognito auth mode
```javascript
                    Amplify.configure({
                        aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS'
                    });
```
But that's Javascript. It would be nice to have an easy way to switch auth modes while your program is running with similar code in Dart.

**Comment by dnys1** [comment]
This feature is available in preview as `0.4.4-rc.1`. To make use of it, pass in `authModeStrategy: AuthModeStrategy.multiAuth` to the constructor of the DataStore plugin.

**Comment by bschulz87** [comment]
While setting the authModeStrategy using 0.4.4-rc.1 is possible, having different authorizationType for the same endpoint in amplifyconfig api.plugins.awsAPIPlugin does not work.

```
const amplifyconfig = ''' {
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "api": {
        "plugins": {
            "awsAPIPlugin": {
                "api_AWS_IAM": {
                    "endpointType": "GraphQL",
                    "endpoint": "https://id.appsync-api.eu-central-1.amazonaws.com/graphql",
                    "region": "eu-central-1",
                    "authorizationType": "AWS_IAM"
                },
                "api_AWS_LAMBDA": {
                    "endpointType": "GraphQL",
                    "endpoint": "https://id.appsync-api.eu-central-1.amazonaws.com/graphql",
                    "region": "eu-central-1",
                    "authorizationType": "AWS_LAMBDA"
                }
            }
        }
    },
```

> Caused by: ApiException{message=There is more than one API configured for this plugin with matching endpoint type., cause=null, recoverySuggestion=Please specify the name of API to invoke in the API method.}


Only using AWS_LAMBDA or AWS_IAM does work.

**Comment by dnys1** [comment]
This has been released in 0.5.0.

@bschulz87 can you confirm this issue is still present, and if so, would you be willing to copy what you have into a new ticket so we can assign it and look into it?
